RIOT-OS/RIOT
# kinetis/fcfield-check*: merge into single file and support binary flash

### Contribution description



This PR merges the two existing scripts to check for the fcfield contents on kinetis boards. It also adds to the merged scipt the capacity of checking for bin files.

Regarding flashing at an offset, since the `fcfield` content is only read on reset, we don't care for its content when flashing an Image after 0x410 or 1040, therefore we won't check its content.

### Testing procedure

Modify any value in the fcfield:

 https://github.com/RIOT-OS/RIOT/blob/8fe12bc82cfbb83a90efd018e11c044d0a40696b/cpu/kinetis/fcfield.c#L55-L73

Flash an elf, hex and bin file and see that it is properly checked:

* `make -C examples/hello-world/ BOARD=frdm-kw41z flash`

```
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf:

/home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf:     file format elf32-littlearm

Contents of section .fcfield:
 0400 ffffffff ffffffff aaffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
```

* `FFLAGS='flash $(HEXFILE)' make -C examples/hello-world/ BOARD=frdm-kw41z flash`

```
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex:

/home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex:     file format ihex

Contents of section .sec1:
 0400 ffffffff ffffffff aaffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
```

* `FFLAGS='flash $(BINFILE)' make -C examples/hello-world/ BOARD=frdm-kw41z binfile flash`

### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin:

```
/home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin:     file format binary

Contents of section .data:
 0400 ffffffff ffffffff aaffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1

```
Remove the changes to fcfield, and repeat the above procedure, no errors will be encountered.




### Issues/PRs references



Used by #11562
